src/notifications/analytics.ts

```diff
@@ -29,3 +35,98 @@ export const trackChangedNotificationSettings = (
     action,
   });
 };
+
+export const trackWalletsSubscribedForNotifications = () => {
+  const walletsStringValue = notificationSettingsStorage.getString(
+    WALLET_TOPICS_STORAGE_KEY
+  );
+  const groupsStringValue = notificationSettingsStorage.getString(
+    WALLET_GROUPS_STORAGE_KEY
+  );
+
+  if (!walletsStringValue || !groupsStringValue) return;
+
+  const wallets = JSON.parse(
+    walletsStringValue
+  ) as WalletNotificationSettings[];
+  const groups = JSON.parse(groupsStringValue) as GroupSettings;
+
+  const { imported, watched } = countWalletsWithNotificationsTurnedOn(wallets);
+
+  trackNumberOfWalletsWithNotificationsTurnedOn(
+    groups[NotificationRelationship.OWNER] ? imported : 0,
+    groups[NotificationRelationship.WATCHER] ? watched : 0
+  );
+};
+
+export type NotificationSubscriptionChangesListener = {
+  remove: () => void;
+};
+
+export const registerNotificationSubscriptionChangesListener = (): NotificationSubscriptionChangesListener => {
+  return notificationSettingsStorage.addOnValueChangedListener(key => {
```

Comment:
Do we really need to listen for changes to the storage instance instead of calling a shared util in the places where a user updates settings? Similar to the handling of `trackChangedNotificationSettings`

Reply:
Code pugson wrote was based on the same thing + it was QA'd I figured out it would be okay to use it since it was the cleanest solution to the problem.

But I also have concerns about the react-native-mmkv library in general